asset/js/setup.js

```diff
@@ -9,6 +9,13 @@ function setup() {
   VueStrap.installEvents(vm);
 }
 
+let baseUrl = window.location.origin;
+
+// environment check
+if (window.location.href.includes('github.io')) {
+  baseUrl = `${baseUrl}/website`;
```

Comment:
Hmm, that's hardcoding the repository name...

Reply:
What about splitting the `window.location.pathname` and taking the first result? 

e.g.
```javascript
if (window.location.href.includes('github.io')) {
baseUrl = `${baseUrl}/${window.location.pathname.split('/')[1]}`; // http://username.github.io/reponame
}
```

Reply:
Better. Probably a stricter check:

```js
if (window.location.origin.endsWith('github.io')) {
  const repository = window.location.pathname.split('/')[1];
  baseUrl += '/' + repository;
}
```

But this still wouldn't work for sites hosted elsewhere. Could we use nunjucks render when copying this asset?

@Gisonrg What do you think?

Reply:
That's true. I suppose it would be possible, so all the instances of baseUrl will be replaced by the author's specified URL? 

After which, we can use `window.location.origin` and concatenate with the baseUrl.

Reply:
Yes.

Actually, `window.location.origin` prefix is unnecessary:

```js
let baseUrl = '';

// Support gh-pages
if (window.location.origin.endsWith('github.io')) {
  const repository = window.location.pathname.split('/')[1];
  baseUrl = `/${repository}`;
}
```